src/server.js

```diff
@@ -105,46 +137,245 @@ server.get('/forgot-password', async(req, res) => {
     res.render('forgotPassword.ejs')
 })
 
-
-
-
-
 server.post('/forgot-password', async(req, res) => {
-    
     const { email } = req.body
-    const userInfo = await db.User.findFirst({ where: { email } })
-    //console.log(userInfo) //for debugging
-    if (!userInfo) {
-        req.flash("error", "There is no account associated with that email")
+
+    const user = await db.User.findFirst({ where: { email } })
+
+    if (!user) {
+        req.flash("error", "Email is not registered")
         res.redirect("/forgot-password")
+            //res.send('User is not registered')
     }
 
+    //user exists and now creating a one time link that is valid for only 15 minutes
+    else {
+        const secret = JWT_SECRET + user.password
+        try {
+            const payload = {
+                email: user.email,
+                id: user.id
+            }
+            const token = jwt.sign(payload, secret, { expiresIn: '15m' })
+            const resetLink = `localhost:8080/reset-password/${user.id}/${token}`
```

Comment:
This won't work if it's hosted

Reply:
added ability to grab publicIp address, .js file is stored in utils, this will require a bit more work though, depending if application is hosted behind a NAT, added todo comment tracking this